Mandater/Controllers/ElectionController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Mandater.Data;
+using Mandater.Models;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Mandater.Controllers
+{
+    [EnableCors("CorsPolicy")]
+    [Produces("application/json")]
+    [Route("api/v1.0.0/")]
+    public class ElectionController : Controller
+    {
+        private readonly ElectionContext _context;
+        private readonly ILogger _logger;
+        public ElectionController(ElectionContext context, ILogger<ElectionController> logger)
+        {
+            _context = context;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Default path method that returns a shallow Country object, showing which countries the API has data on.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("{deep?}")]
+        public IEnumerable<Country> GetCountries(bool deep = false)
+        {
+            if (deep)
+            {
+                return _context.Countries
+                        .Include(c => c.ElectionTypes)
+                            .ThenInclude(c => c.Elections)
+                                .ThenInclude(c => c.Results);
+            }
+            return _context.Countries;
+        }
+
+        [HttpGet("{countryCode}")]
+        public IEnumerable<ElectionType> GetElectionTypes(string countryCode)
```

Comment:
This method crashes when trying to execute [HttpGet("{deep?}")].
Needs to be fixed before merge.

Reply:
It doesn't crash for me, are you sure this is the method that is crashing?